### PR TITLE
Expand SWA Tutorial

### LINKED ISSUE #2492: About the SWA tutorial
Given that SWA needs to have access to a particular LR schedule, I think it might be a good idea to actually update the [tutorial](https://www.tensorflow.org/addons/tutorials/average_optimizers_callback) with a minimal example of incorporating such schedules. 

In my experience (as well as what the original authors assert), in order for someone to reap the benefits of SWA, the schedule is important.

### PR DESCRIPTION
**Pull request by jonpsy**
# Description

- [x] Fix the mkdir for the training path
- [x] Use different folder to store checkpoints for each optimizer.
- [x] Use cyclic averaging for SWA Example. 
- [x]  Stocastic spelling error 

Fixes #2492

## Type of change

- [x] Bug fix
- [x] Updated or additional documentation

# Checklist:

- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

@bhack Kindly review

### CODE REVIEW

**Comment by jonpsy** [comment]
The notebook was using the same folder to store the weights of for each optimizer. This lead to **loading the same weights** for each optimizer. I fixed that issue along with the requested change in the original issue, hopefully this gets reviewed.

**Comment by jonpsy** [comment]
Can someone help with the ```google:second person``` error please?

**Comment by jonpsy** [comment]
@sayakpaul  Could you help review this? thank you

File: docs/tutorials/average_optimizers_callback.ipynb [hunk 1]
```diff
@@ -76,9 +76,9 @@
         "\n",
```
Comment (sayakpaul) [inline docs/tutorials/average_optimizers_callback.ipynb:76]: <p>Might be a good idea to add a note to mention SWA needs to have an LR schedule in order to get to the center of a loss basin.</p><p> </p><p>Usage of "we" is causing the <code>google:second person</code>  error I think. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/2497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (sayakpaul) [inline docs/tutorials/average_optimizers_callback.ipynb:76]: <pre class="ql-syntax" spellcheck="false">Line #4.    stocastic_avg_sgd = tfa.optimizers.SWA(sgd, 1000, 50)
</pre><p>How is this inducing the LR schedule we talked about in the original issue? For more details, refer to the <a href="https://pytorch.org/blog/stochastic-weight-averaging-in-pytorch/" target="_blank">Stochastic Weight Averaging</a> section of this blog post. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/2497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (sayakpaul) [inline docs/tutorials/average_optimizers_callback.ipynb:76]: <p>I think it would be worth adding a line or two explain how one needs to be careful about setting up the checkpoint paths. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/2497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (sayakpaul) [inline docs/tutorials/average_optimizers_callback.ipynb:76]: <p>Please maintain spaces between <code>#</code> when you are commenting. I would also club together cells that are performing similar actions. For example,</p><p> </p><pre class="ql-syntax" spellcheck="false"># Build and train the model
...

# Load the latest checkpoints into the model
...
</pre><p> </p><p>Also, I am not sure why this line is needed now? <code>model.save_weights(vanilla_checkpoint_path.format(epoch=0))</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/2497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (sayakpaul) [inline docs/tutorials/average_optimizers_callback.ipynb:76]: <pre class="ql-syntax" spellcheck="false">Line #8.    model.save_weights(ma_checkpoint_path.format(epoch=0))
</pre><p>Ditto.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/2497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (jonpsy) [reply docs/tutorials/average_optimizers_callback.ipynb:76]: <p>Thanks, I'll have a look.</p>

Comment (jonpsy) [reply docs/tutorials/average_optimizers_callback.ipynb:76]: <p>I don't think thats neccesary because its only expected from the user and doesn't relate heavily to our main objective here. Thats my opinion atleast.</p>

Comment (sayakpaul) [reply docs/tutorials/average_optimizers_callback.ipynb:76]: I referred to [your comment](https://github.com/tensorflow/addons/pull/2497#issuecomment-855204867):

> The notebook was using the same folder to store the weights of for each optimizer. This lead to loading the same weights for each optimizer.

Comment (jonpsy) [reply docs/tutorials/average_optimizers_callback.ipynb:76]: <p>About the spaces, sure. About format, I don't exactly know as to why it's done, I've used this as reference https://www.tensorflow.org/tutorials/keras/save_and_load. I'll try to remove and see it's effect, if its unnecessary I'll remove it :+1: </p><p> </p>

Comment (jonpsy) [reply docs/tutorials/average_optimizers_callback.ipynb:76]: <p>I see, i was a bit confused on that part thanks for sharing the blog. About "we", I think I've removed every instance of it and yet it throws error....</p>

Comment (jonpsy) [reply docs/tutorials/average_optimizers_callback.ipynb:76]: Yes, that was an error I fixed.

**Comment by jonpsy** [comment]
Thanks for all the reviews, I'll try and apply that suggestion.

File: docs/tutorials/average_optimizers_callback.ipynb (hunk 1 above)
Comment (sayakpaul) [reply docs/tutorials/average_optimizers_callback.ipynb:76]: Yeah, so chances are that users won't notice this. So, explicitly making it clear inside the tutorial is helpful.

Comment (sayakpaul) [reply docs/tutorials/average_optimizers_callback.ipynb:76]: @MarkDaoust what could be the reason?

**Comment by sayakpaul** [comment]
@jonpsy it might be even better to also include a step to update the batch norm statistics after the SWA model variant is trained since that is recommended. From the [docs](https://www.tensorflow.org/addons/api_docs/python/tfa/optimizers/SWA):

> Note: If your model has batch-normalization layers you would need to run the final weights through the data to compute the running mean and variance corresponding to the activations for each layer of the network. From the paper: If the DNN uses batch normalization we run one additional pass over the data, to compute the running mean and standard deviation of the activations for each layer of the network with SWA weights after the training is finished, since these statistics are not collected during training. For most deep learning libraries, such as PyTorch or Tensorflow, one can typically collect these statistics by making a forward pass over the data in training mode (Averaging Weights Leads to Wider Optima and Better Generalization)

File: docs/tutorials/average_optimizers_callback.ipynb (hunk 1 above)
Comment (MarkDaoust) [reply docs/tutorials/average_optimizers_callback.ipynb:76]: " We will use default params here" on line 223

**Comment by bhack** [comment]
Thanks for the contribution. We are mainly collecting more examples using SWA in keras.io:

https://github.com/search?q=org%3Akeras-team+swa&type=code

Please open a PR there.
